cli/azd/internal/repository/app_init.go

```diff
@@ -472,23 +485,74 @@ func (i *Initializer) prjConfigFromDetect(
 					config.Resources["postgres"] = &project.ResourceConfig{
 						Type: project.ResourceTypeDbPostgres,
 						Name: spec.DbPostgres.DatabaseName,
+						Props: project.PostgresProps{
+							DatabaseName: spec.DbPostgres.DatabaseName,
+							AuthType: decideAuthType(
+								spec.DbPostgres.AuthUsingManagedIdentity,
+								spec.DbPostgres.AuthUsingUsernamePassword,
+								false),
+						},
```

Comment:
How about do like this:

```
type AuthType string

const (
	UserAssignedManagedIdentity AuthType = "User assigned managed identity"
	UsernamePassword AuthType = "Username and password"
	ConnectionString AuthType = "Connection String"
)
```
Then delete all bool variables like `AuthUsingManagedIdentity`, use `AuthType` instead?

Reply:
The auth type value is output to the `azure.yaml`, which means it should be like enum value, for users to set in the yaml.

Reply:
And yes, we could delete all the bool variables.